src/main/database.cpp

```diff
@@ -275,12 +275,21 @@ void DatabaseInstance::Initialize(const char *database_path, DBConfig *user_conf
 
 	create_api_v0 = CreateAPIv0Wrapper;
 
-	if (user_config && !user_config->options.use_temporary_directory) {
+	db_file_system = make_uniq<DatabaseFileSystem>(*this);
+
+	// resolve the type of the database we are opening
+	auto &fs = FileSystem::GetFileSystem(*this);
+	DBPathAndType::ResolveDatabaseType(fs, config.options.database_path, config.options.database_type);
+
+	if (config.options.temporary_directory.empty()) {
+		config.SetDefaultTempDirectory();
+	}
+
+	if (!config.options.use_temporary_directory) {
 		// temporary directories explicitly disabled
 		config.options.temporary_directory = string();
 	}
 
```

Comment:
```suggestion
	if (!config.options.use_temporary_directory) {
		// temporary directories explicitly disabled
		config.options.temporary_directory = string();
	}

	if (config.options.temporary_directory.empty()) {
		config.SetDefaultTempDirectory();
	}
```
I think it's somewhat better to invert the ifs, so both to mirror current code, and avoid calling SetDefaultTempDirectory (that might also have side effects, even though it doesn't at the moment).

Reply:
Thanks Carlo. I adopted your suggestion with a small change in https://github.com/duckdb/duckdb/pull/14401/commits/e5aadfb8f94e43c5f1f3b7478aaffd73e3b1eef4.